_posts/2023-02-16-announcing-ballerina-2201.4.0-swan-lake-update-4.md

```diff
@@ -0,0 +1,51 @@
+- The `graphql` package is improved such that now, a single client can have multiple subscriptions (multiplexing), and if additional information about a particular field of a GraphQL document (e.g., subfields of a field) is required, that information can be accessed inside a resolver or an interceptor method.
+- A new JetStream client is introduced to the `nats` package to carry out the publishing and subscribing streaming functionalities related to the NATS JetStreams feature.
+- The `constraint` package is enhanced by adding constraint validation support for read-only types.
+- The `oauth2` package now allows the use of inferred values for refresh token configurations through the password grant type, and string values for the `scopes` field in the client grant configuration.
+
+## Code to cloud updates
+
+Now, you can disable creating thin JAR files when generating the Docker image in order to reduce the build time of the generation. However, this will disable the Docker layering feature.  Also, the `CMD` property of the Dockerfile is improved to support complex package names.
+
+## Developer tools updates
+
+- This Update 4 release introduces a new `Create variable with Type` code action along with `Quick Pick` support for selecting expressions in the `Extract to constant` code action. Additionally, the `Rename popup` feature was added to the `Extract to constant`, `Extract to local variable`, and 'Extract to function' code actions. 
+- From this release onwards, you can generate Ballerina client and service declarations from Swagger 2.0 (i.e., OpenAPI 2.0) definitions along with a few other improvements done for OpenAPI object schema attributes and the default request and response types of the generated Ballerina resource/remote methods.
+- New options are introduced to the `bal format` command to format modules or single files selectively with a provided Ballerina package. Also, the `--native` flag is introduced to the `bal test` command for executing tests using a GraalVM native executable image. This is an experimental feature.
+- The GraphQL tool can now generate GraphQL SDL schema files from GraphQL services for users to share those files for documentation or any other purpose as required.
+
+## Ballerina packages updates
+
+From this Update 4 release onwards, now, you can maintain code generated by other tools in a Ballerina package. This feature allows you to keep the generated source code in a separate directory named ‘generated’ within the package structure while treating those files as part of the Ballerina package source.
```

Comment:
Shall we recheck if we really need "now" wherever it has been used? Redundant here since we start with "From this Update 4 release onwards"?

Actually even that sentence has both this and update 4, one is kinda redundant?

```suggestion
From this release onward, you can maintain code generated by other tools in a Ballerina package. This feature allows you to keep the generated source code in a separate directory named ‘generated’ within the package structure while treating those files as part of the Ballerina package source.
```

Reply:
Updated.